tezos-encoding/src/nom.rs

```diff
@@ -572,6 +573,56 @@ where
     }
 }
 
+#[derive(Debug, PartialEq)]
+/// Lazy, memoized deserialization of a dynamic block.
+pub struct LazyCell<'a, T>
+where
+    T: NomReader<'a>,
+{
+    // Invariant: when the `deserialized` field is some, it's the
+    // result of deserialization of the `serialized` field using
+    // dynamic(T::nom_read)
+    serialized: &'a [u8],
+    deserialized: Option<T>,
+}
+
+impl<'a, T> NomReader<'a> for LazyCell<'a, T>
+where
+    T: NomReader<'a>,
+{
+    fn nom_read(input: &'a [u8]) -> NomResult<'a, Self> {
+        map(dynamic(rest), |serialized| LazyCell {
+            serialized,
+            deserialized: None,
+        })(input)
+    }
+}
+
+impl<'a, T> LazyCell<'a, T>
+where
+    T: NomReader<'a>,
+{
+    pub fn to_bytes(self) -> &'a [u8] {
+        self.serialized
+    }
+
+    pub fn has_been_forced(&self) -> bool {
```

Comment:
why is this needed?
If it's really needed, you could have a `fn value(&self) -> Option<&T>`?

Reply:
I don't know if we will ever need this but the corresponding feature in OCaml is needed in the protocol to carbonate lazy (de)serialization.

Let's remove it for now.